Library.py

```diff
@@ -0,0 +1,156 @@
+from abc import ABC, abstractmethod
+from pydantic import BaseModel
+
+
+class Publication(BaseModel):
+    title: str
+    author: str
+    year: int
+
+
+
+class BookModel(Publication):
+    def display_info(self):
+        return f'{self.title} - {self.author} from {self.year} year'
+
+class MagazineModel(Publication):
+    month: str
+    def display_info(self):
```

Comment:
display_info is redundant, because you have defined it below for AbstractPublication and Book